src/core/shared_state/currentMapPosition.ts

```diff
@@ -1,70 +1,77 @@
 export type CenterZoomPosition = {
-  // type: 'centerZoom';
   lat: number;
   lng: number;
   zoom: number;
 };
 export type Bbox = [number, number, number, number];
 export type BboxPosition = {
-  // type: 'bbox';
   bbox: Bbox;
 };
 
 export type MapPosition = CenterZoomPosition | BboxPosition;
 
 export type CurrentMapPositionAtomState = MapPosition | null;
 
+// TODO: #20160 update currentMapPositionAtom to reatom v3
 export const currentMapPositionAtom = createAtom(
   {
     setCurrentMapPosition: (mapPosition: { lat: number; lng: number; zoom: number }) =>
       mapPosition,
     setCurrentMapBbox: (mapBbox: Bbox | [[number, number], [number, number]]) => mapBbox,
+    updateCurrentMapPosition: (mapPosition: { lat: number; lng: number; zoom: number }) =>
+      mapPosition,
+    currentMapAtom,
   },
   ({ onAction }, state: CurrentMapPositionAtomState = null) => {
-    onAction('setCurrentMapPosition', (position) => {
-      if (state === null || !('lng' in state)) {
-        state = position;
-      } else {
-        const { lat, lng, zoom } = position;
-        if (
-          'lng' in state &&
-          (state.lat !== lat || state.lng !== lng || state.zoom !== zoom)
-        ) {
-          state = position;
+    function jumpTo(map: Map, position: MapPosition) {
+      const { lng, lat, zoom } = position as CenterZoomPosition;
+      requestAnimationFrame(() => {
+        map?.stop();
+      });
+      setTimeout(() => {
+        const mapCenter = map.getCenter();
+        const mapZoom = map.getZoom();
+        if (mapCenter.lng !== lng || mapCenter.lat !== lat || mapZoom !== zoom) {
+          requestAnimationFrame(() => {
+            map?.jumpTo({
+              center: [lng, lat],
+              zoom: zoom,
+            });
+          });
         }
+      }, 100);
+    }
+
+    onAction('setCurrentMapPosition', (position) => {
+      const map = currentMapAtom.getState();
+      if (map) {
+        jumpTo(map, position);
       }
+      state = position;
     });
 
     onAction('setCurrentMapBbox', (bbox) => {
-      const position = { bbox: bbox.flat() as Bbox };
-      const prev = state;
-      if (prev === null || !('bbox' in prev)) {
-        state = position;
-      } else {
-        if (
-          'bbox' in prev &&
-          prev?.bbox?.some((coord: number, i: number) => coord !== position.bbox[i])
-        ) {
-          state = position;
-        }
-      }
+      let position = { bbox: bbox.flat() } as MapPosition;
       const map = currentMapAtom.getState();
-      if (!map) return;
-      const cam = getCameraForBbox(bbox, map);
-      if (cam.center && 'lng' in cam.center) {
-        const { zoom } = cam;
-        const { lat, lng } = cam.center;
-        if (
-          prev == null ||
-          ('lng' in prev && (prev.lat !== lat || prev.lng !== lng || prev.zoom !== zoom))
-        ) {
-          state = { ...position, lat, lng, zoom };
+      if (map) {
+        const cam = getCameraForBbox(bbox, map);
```

Comment:
Not `cam` but `viewport`

Reply:
@dqunbp  I can't find, where is the concept of viewport coming from here? 
`getCameraForBbox` uses Maplibre's `map.cameraForBounds()` and returns CenterZoomBearing object

I never heard about Center+Zoom+Bearing being called a Viewport before, sounds counterintuitive (doesn't Viewport usually describe the portion of the screen). Where can I read about it?

Reply:
As for the variable name, it's in line with how Maplibre calls it (the function uses `map.cameraForBounds()` which returns CenterZoomBearing). Since it's not a Viewport with a full set info, maybe it would make sense to just call it `centerZoomBearing`?

Reply:
There are similar variable called "camera" across the code base. Maybe let's keep it like this for now, for the sake of consistency?